go/controller/autoscaler/autoscaler.go

```diff
@@ -62,7 +63,6 @@ func NewAutoscaler(cluster Cluster, options ...func(*Autoscaler)) *Autoscaler {
 		cluster: cluster,
 		ticker:  time.NewTicker(defaultLoopDur),
 		jobs:    make(map[string]job),
-		mutex:   &sync.Mutex{},
```

Comment:
Don't need to initialize `sync.Mutex`, but need to initialize `*sync.Mutex`. People typically use `sync.Mutex`.